controllers/glanceapi_controller.go

```diff
@@ -522,6 +522,9 @@ func (r *GlanceAPIReconciler) generateServiceConfigMaps(
 		templateParameters["ShowMultipleLocations"] = false
 	}
 
+    // Configure glance image import internal plugins, default to 'no_op'
+    templateParameters["GlanceImageImportPlugins"] = ['no_op']
```

Comment:
I think this is a problem for golang, and this should actually be changed into something different.
The goal here is to render a string like "['no_op']" that, eventually, can be improved as long as plugins are enabled.
If you would like to give this default, a way would be having a function (in `pkg/glanceapi/`) like this:

```
func render_plugins() {

	var res string;
	var plugins = []string {
		"no_op",
		"p1",
		"p2",
	}

	res = "["
	for _, p := range plugins {
		res += fmt.Sprintf("'%s',", p)
	}

	strings.TrimSuffix(res, ",")
	res += "]"
	fmt.Println(res)
}
```

and assign to `templateParameters["GlanceImageImportPlugins"]` the result of this function.
However, if this is the default parameter we'd like to enforce, another and maybe better approach would just be removing this line and add the following to `glance-image-import.conf`:
```
[image_import_opts]
image_import_plugins = ['no_op']
```
and see if an override using `customServiceConfig` in the main `glance-api.conf` will work.
The point is: do we want the `glance-operator` being smart enough to `auto-enable` the right plugins?
If we have a set of values we always want to set we can just follow up on this change and change the `image_import_plugins` accordingly in the future, but for now enforcing ['no_op'] in the file should work.

Reply:
Yes, we want glance-operator to auto-enabled 'image_conversion' plugin if we have configured ceph as glance backend. The motive of this patch is to introduce glance-image-import.conf with default no_op plugin and then modify it to set image_conversion related config options in the glance-image-import.conf

We also have other plugins such as;
inject_imgae_metadata, image_conversion but these plugins are totally deployment/customer specific. (i.e. to be enabled only if customer want to use it)

If you still think just setting ['no_op'] directly in conf file initially then I will modify the pull request.

Reply:
> Yes, we want glance-operator to auto-enabled 'image_conversion' plugin if we have configured ceph as glance backend.

Yeah, I had something like that in mind: however, we first need to be able to "detect" that Ceph is enabled, and this can be done by inspecting the `extraMounts` (the `extraVolType` parameter) as well as parsing the `enabled_backends` (which is not part of this PR).

>The motive of this patch is to introduce glance-image-import.conf with default no_op plugin and then modify it to set image_conversion related config options in the glance-image-import.conf

We can set/enable `image_conversion` if Ceph is enabled, but this is something we can't easily do atm
> 
> We also have other plugins such as; inject_image_metadata, image_conversion but these plugins are totally deployment/customer specific. (i.e. to be enabled only if customer want to use it)

So I suspect we want the customer (human operator) being able to customize/override this option: am I understanding it correctly? 
In this case, we can `no_op` by default and rely on overrides, which is easier to handle.
If we go with this approach, we don't have to "detect" that Ceph is the backend, because it's the human operator, in the first place, that inject the Ceph bits (so he's supposed to be aware) and we should document how the following:

```
[image_import_opts]
image_import_plugins = ['no_op']
```
should be customized.

> 
> If you still think just setting ['no_op'] directly in conf file initially then I will modify the pull request.
Yeah, I think it's cleaner for now, and we can focus on finding a way to use `customServiceConfig` to override that section.
If the user provides something like:

```
apiVersion: glance.openstack.org/v1beta1
kind: Glance
metadata:
  name: glance
spec:
  serviceUser: glance
  ...
  customServiceConfig: |
    [DEFAULT]
    enabled_backends=default_backend:rbd,ceph1:rbd
    enabled_backends=default_backend:rbd
    [glance_store]
    default_backend=default_backend
    [default_backend]
    rbd_store_ceph_conf=/etc/ceph/ceph.conf
    rbd_store_user=admin
    rbd_store_pool=images
    store_description=Ceph glance store backend.
  databaseInstance: openstack
 ...
 ...
  extraMounts:
    - name: v1
      region: r1
      extraVol:
          extraVolType: Ceph
          volumes:
          - name: ceph
            projected:
              sources:
              - secret:
                  name: ceph-conf-files
          mounts:
          - name: ceph
            mountPath: "/etc/ceph"
```

we should try to inject the customized parameter with the Ceph bits.
Let me know your thoughts about this!

Reply:
FWIW I'd add the following directly to glance-image-import.conf:
```
[image_import_opts]
image_import_plugins = ['no_op']
```
and then have an expectation that the user would override the `image_import_plugins` within the `customServiceConfig`. The same way we expect the user to override `customServiceConfig` if they want to configure a backend like Ceph. Otherwise we go down a long road and write a lot of code to have some layer of abstraction on top of standard Glance config files which seems to simplify in the beginning of the project but adds complexity over time.